File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum n = 
    let gauss_sum = ((n) * (n + 1)) / 2 in
    gauss_sum * gauss_sum

let sum_of_squares n =
    ((n) * (n + 1) * (2 * n + 1)) / 6

let difference_of_squares n =
    (square_of_sum n) - (sum_of_squares n)

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int square_of_sum(unsigned int number) {
  int sum_of_N = (number * (number + 1)) / 2;
  return sum_of_N * sum_of_N;
}

unsigned int sum_of_squares(unsigned int number) {
  return (number * (number + 1) * (2 * number + 1)) / 6;
}

unsigned int difference_of_squares(unsigned int number) {
  return square_of_sum(number) - sum_of_squares(number);
}
